### ISSUE TITLE
Issues with utteranceQueue collapsing Utterances of the same type

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/scenery-phet#405]
1) `addToFront` doesn't support this feature at all at the moment. This just seems like an oversight.
2) The way to create alerts of the same type is called `typeId`. This is a horrible name, we should rename this to better indicate what this is doing.

For 1, I think the behavior should be that if an Utterance is added to the front of the queue with a type, all other utterances in the queue with the same type should be removed. Does this seem correct?

For 2, any ideas for a better name for this? Here is potential doc.

{number|string|null} - Adds a signifier to the utterance that prevents too many alerts of the same type spamming the queue. If more than one Utterance with the same {{nameHere}} is added to the queue, all of the other utterances of the same type that were previously added were removed. If null, this feature is disabled and all will be announced.

`preventDuplicatesId`
`preventDuplicatesName`
`collapseTypeName`
`collapseGroup`
`collapseId`
`utteranceType`
`utteranceGroup`
`utteranceTypeName`
`preventSpamId`

I like `utteranceGroup` or `collapseGroup`, @mbarlow12 @zepumph any other votes?

**Comment by zepumph** [comment]
I said to @jessegreenberg on slack:

```
Michael Kauzmann [10:27]
This may help with finding a name for typeID
https://stackoverflow.com/questions/2319086/a-queue-that-ensure-uniqueness-of-the-elements
Stack Overflow
A Queue that ensure uniqueness of the elements?
I'm looking for a implementation of java.util.Queue or something in the Google collection who behave like a Queue, but also ensure that each element of the queue is unique. (all further insertion w...
I would recommend `ID` rather than `Id`. What about something with `unique` (edited)
`classIDToEnsureUniqueness`
`uniqueGroupingID`
`uniqueClassification`
`uniqueClassificationID`
```

**Comment by jessegreenberg** [comment]
I like `unique` and `ID`, Maybe `set` to indicate that items in the set specified by the ID because by definition a set can have no duplicates? @zepumph and @jessegreenberg don't like this because it indicates that there is a subset that you can manipulate or that maybe the queue itself is a 'set'.

**Comment by jessegreenberg** [comment]
Lets go with `uniqueGroupId` for now.

**Comment by jessegreenberg** [comment]
This was handled in the above commits, @mbarlow12 could you please review this? As part of the review could you please verify that this fixes the alert repetition that you reported in faradays-law?

**Comment by jessegreenberg** [comment]
Thanks @mbarlow12 sounds good. But I am also curious about why it is still sub-optimal in FL. Ill make an issue there to learn more.
